src/graph.py

```diff
@@ -107,3 +107,136 @@ def neighbors(self, node):
             raise KeyError("{} isn't a node in the graph".format(node))
         else:
             return list(node.edges)
+
+    def breadth_first_traversal(self, start):
+        """Return the full visited path when breadth traversal is complete after starting with any node."""
+        self.is_node(start)
+        traversal = []
+        traversal.append(start)
+        index = 0
+        while True:
+            try:
+                traversal.extend([x for x in traversal[index].edges if x not in traversal])
+            except IndexError:
+                break
+            index += 1
+        return traversal
+
+    def depth_first_traversal(self, start):
+        """Return the full visited path when depth traversal is complete after starting with any node."""
+        self.is_node(start)
+        traversal = []
+        traversal = self.depth_traversal_add_node(start, traversal)
```

Comment:
since your `depth_traversal_add_node` function modifies the traveral list in place, it doesn't need to return this list.